presto-main/src/main/java/com/facebook/presto/sql/SqlPath.java

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.String.format;
+
+public class SqlPath
+{
+    private List<SqlPathElement> parsedPath;
+    private String rawPath;
+
+    @JsonCreator
+    public SqlPath(@JsonProperty("rawPath") String path)
+    {
+        this.rawPath = path;
+    }
+
+    @JsonProperty
+    public String getRawPath()
+    {
+        return rawPath;
+    }
+
+    public List<SqlPathElement> getParsedPath()
+    {
+        if (parsedPath == null) {
+            parsePath();
+        }
+
+        return parsedPath;
+    }
+
+    private void parsePath()
+    {
+        //TODO: add further verification once default path is determined
+        ImmutableList.Builder<SqlPathElement> builder = ImmutableList.builder();
+        //avoid splitting on commas contained within quotes as part of catalog/schema names
+        String[] catalogAndSchemas = rawPath.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
+
+        for (String pathEntry : catalogAndSchemas) {
+            //avoid splitting on periods contained within quotes as part of catalog/schema names
+            String[] catalogAndSchema = pathEntry.split("\\.(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
+
+            if (catalogAndSchema.length > 2 || catalogAndSchema.length < 1) {
+                throw new IllegalArgumentException(format("Schema name [%s] is invalid", pathEntry));
+            }
+            if (catalogAndSchema.length == 2) {
+                builder.add(new SqlPathElement(catalogAndSchema[0], catalogAndSchema[1]));
+            }
+            if (catalogAndSchema.length == 1) {
+                builder.add(new SqlPathElement(null, pathEntry));
+            }
+        }
+
+        this.parsedPath = builder.build();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if ((obj == null) || (getClass() != obj.getClass())) {
+            return false;
+        }
+        SqlPath that = (SqlPath) obj;
+        return Objects.equals(parsedPath, that.parsedPath);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(parsedPath);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("rawPath", rawPath)
+                .add("parsedPath", parsedPath != null ? Iterables.toString(parsedPath) : null)
```

Comment:
Nit: if the paths contain commas it might be clearer to re-escape the quotes, quote each path element and join with commas.